xbmc/platform/android/storage/AndroidStorageProvider.cpp

```diff
@@ -451,3 +448,41 @@ bool CAndroidStorageProvider::GetStorageUsage(const std::string& path, std::stri
       PATH_MAXLEN, totalSize, "G", usedSize, "G", freeSize, "G", usedPercentage, "%");
   return true;
 }
+
+bool CAndroidStorageProvider::GetExternalStorage(std::string& path,
+                                                 const std::string& type /* = "" */)
+{
+  std::string sType;
+  std::string mountedState;
+  bool mounted = false;
+
+  if (type == "files" || type.empty())
+  {
+    CJNIFile external = CJNIEnvironment::getExternalStorageDirectory();
```

Comment:
maybe take the opportunity to split this small block of code into its own method? The logic is duplicated here and near the end of the method

Reply:
The two "external" variables are necessary, since it's not possible to use the CJNIFile() constructor to have a single external variable that would simplify the code.

The following code could be extracted to a private function:
https://github.com/xbmc/xbmc/blob/deb59d83f653da318521628518a4f1d752cb766f/xbmc/platform/android/storage/AndroidStorageProvider.cpp#L462-L463
but it doesn't improve much, I think I'd leave it as it was originally.